core-library/src/main/java/org/silverpeas/core/index/search/model/SearchResult.java

```diff
@@ -0,0 +1,215 @@
+package org.silverpeas.core.index.search.model;
+
+import org.silverpeas.core.contribution.contentcontainer.content.GlobalSilverContent;
+import org.silverpeas.core.i18n.AbstractI18NBean;
+import org.silverpeas.core.util.StringUtil;
+import org.silverpeas.core.util.file.FileServerUtils;
+
+import java.time.LocalDate;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * @author Nicolas Eysseric
+ */
+public class SearchResult extends AbstractI18NBean {
+
+  private String keywords;
+  private LocalDate creationDate;
+  private String creatorId;
+  private LocalDate lastUpdateDate;
+  private String lastUpdaterId;
+
+  private String id;
+  private String componentId;
+  private String type;
+
+  private float score;
+  private String serverName;
+  private String thumbnailURL;
+  private String attachmentFilename;
+  private boolean externalResult = false;
+  private boolean alias = false;
+
+  /**
+   * list of XML form fields used to sort results
+   */
+  private Map<String, String> sortableXMLFormFields = null;
+  private Map<String, String> formFieldsForFacets;
+
+  public static SearchResult fromIndexEntry(MatchingIndexEntry mie) {
+    return new SearchResult(mie);
+  }
+
+  public static SearchResult fromGlobalSilverContent(GlobalSilverContent gsc) {
+    return new SearchResult(gsc);
+  }
+
+  private SearchResult(MatchingIndexEntry mie) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1213)